src/index.js

```diff
@@ -97,7 +98,12 @@ function isTabbable(node) {
 }
 
 function isNodeMatchingSelectorFocusable(node) {
-  if (node.disabled || isHiddenInput(node) || isHidden(node)) {
+  if (
+    node.disabled ||
+    isHiddenInput(node) ||
+    isHidden(node) ||
+    isDetailsWithSummary(node)
```

Comment:
I think upon reading this for the first time, this will seem counter-intuitive to a reader who isn't familiar with the nuance of details with/out summary inside. Could you add a comment here about the fact that when it contains a summary, it's the summary that's focusable, not the details? (Even I had to ponder about this one just now!)